File: sdk/php/src/Resources/Contacts.php
<?php

namespace HiTechCloud\SDK\Resources;

class Contacts extends BaseResource
{
    public function list(array $params = []): array
    {
        return $this->http->get('/api/contacts', $params);
    }

    public function get(int $contactId): array
    {
        return $this->http->get("/api/contacts/{$contactId}");
    }

    public function create(array $data): array
    {
        return $this->http->post('/api/contacts', $data);
    }

    public function update(int $contactId, array $data): array
    {
        return $this->http->put("/api/contacts/{$contactId}", $data);
    }

    public function delete(int $contactId): array
    {
        return $this->http->delete("/api/contacts/{$contactId}");
    }
}

File: sdk/csharp/Resources/Contacts.cs
namespace HiTechCloud.SDK.Resources;

public class Contacts : BaseResource
{
    public Contacts(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ContactsAsync()
    {
        return await Http.GetAsync("/api/contact");
    }

    public async Task<Dictionary<string, object?>> CreateContactAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/contact", data);
    }

    public async Task<Dictionary<string, object?>> PrivilegesAsync()
    {
        return await Http.GetAsync("/api/contact/privileges");
    }

    public async Task<Dictionary<string, object?>> ContactAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/contact/{id}", id));
    }

    public async Task<Dictionary<string, object?>> UpdateContactAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/contact/{id}", id), data);
    }
}
